LoopKit/Alert.swift

```diff
@@ -22,6 +22,26 @@ public protocol AlertResponder: AnyObject {
     func acknowledgeAlert(alertIdentifier: Alert.AlertIdentifier, completion: @escaping (Error?) -> Void) -> Void
 }
 
+public struct PersistedAlert: Equatable {
+    public let alert: Alert
+    public let issuedDate: Date
+    public let retractedDate: Date?
+    public let acknowledgedDate: Date?
+    public init(alert: Alert, issuedDate: Date, retractedDate: Date?, acknowledgedDate: Date?) {
+        self.alert = alert
+        self.issuedDate = issuedDate
+        self.retractedDate = retractedDate
+        self.acknowledgedDate = acknowledgedDate
+    }
+}
+
+/// Protocol for looking up alerts persisted in storage
+public protocol PersistedAlertStore {
+    /// Look up all issued, but unretracted, alerts for a given `managerIdentifier`.  This is useful for an Alert issuer to see what alerts are extant (outstanding).
+    /// NOTE: the completion function may be called on a different queue than the caller.  Callers must be prepared for this.
+    func lookupOutstandingAlerts(managerIdentifier: String, completion: @escaping (Swift.Result<[PersistedAlert], Error>) -> Void)
```

Comment:
[nit] what about using `lookupUnrestractedAlerts` to keep the terms consistent.

Reply:
Thinking about this more, my use case would be to get all unacknowledged and unretracted alerts to avoid issuing the same alert again when it is detected as `pending`.

Reply:
@nhamming what is your use case?  The one _I_ need for COASTAL-651 is `unretracted` alerts only.  If you need to add `lookupAllUnacknowledgedUnretracted` I can add that one here too but I won't use it...